lib/view_model/active_record/update_operation.rb

```diff
@@ -526,35 +540,121 @@ def build_updates_for_has_many_through(association_data, reference_strings, upda
       previous_through_viewmodels_by_indirect_ref =
         previous_through_viewmodels.group_by(&viewmodel_reference_for_indirect_model)
 
-      new_through_viewmodels = reference_strings.map do |ref|
-        target_reference = update_context.resolve_reference(ref).viewmodel_reference
+      case association_update
+      when ReferencedCollectionUpdate::Replace
+        target_indirect_viewmodels =
+          association_update.values.map do |ref|
+            {
+              ref:           ref,
+              viewmodel_ref: update_context.resolve_reference(ref).viewmodel_reference,
+            }
+          end
+
+      when ReferencedCollectionUpdate::Functional
```

Comment:
Some of the below could really do with some high level comments motivating the particular manipulations.